text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,258 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+I'm going to focus on writing help text that explains the proposed commands. The help text should describe the commands well enough to understand what this RFC is proposing, and define the set of flags that adjust their behaviour.
+
+### push: upload datasets to qri peers
+```
+$ qri push --help
+Push sends datasets to a remote qri node. A push updates the dataset log on the
+remote and sends as many versions as the remote is willing to accept. A remote
+peer determines it's own rules on which versions to accept, but by default
+favour the latest versions of a dataset.
+
+If no remote is specified, qri pushes to the registry. But pushes 
+can just as easily send to another peer that is configured to accept datasets 
+from you. For details on  setting peers for dataset exchange read this doc:
+https://qri.io/docs/hosting-datasets
+
+Usage:
+  qri push [REMOTE] DATASET [DATASET...] [flags]
+
+Aliases:
+  publish
+
+Examples:
+  # push a dataset to the registry
+  $ qri push me/dataset
+
+  # push a specific version of a dataset to the registry:
+  $ qri push me/dataset@/ipfs/QmHashOfVersion
+
+  # push only the latest two versions to a peer named xiao
+  $ qri push xiao me/dataset --revisions 2
+
+Flags:
+  -l, --logs               send only dataset history
+  -r, --revisions          send only the lastest commit data
+```
+
+### pull: download datasets from the qri network
+```
+$ qri pull --help
+Pull downloads datasets and stores them locally, fetching the dataset log and
+dataset version(s). By default pull only fetches the latest version of a dataset
+
+Because pull grabs the latest version by default, it's common to 
+
+If qri connect is running
+
+Usage:
+  qri pull [REMOTE] DATASET [DATASET...] [flags]
+
+Examples:
+  # download a dataset log and latest version
+  $ qri pull b5/world_bank_population
+
+  # download a dataset log and all versions
+  $ qri pull b5/world_bank_population --all
+
+  # pull a specific version from a remote by hash
+  $ qri pull ramfox b5/world_bank_population@/ipfs/QmFoo...
+
+Flags:
+  -l, --logs              download only dataset history
+  -r, --revisions         fetch a set number of versions from the latest commit
+  -a, --all               fetch all versions, overrides revisions
+  -k, --keep-files        when in a working directory, don't update files
+```
+
+`revisions` and `all` flags mimic the way `qri remove` works.
+
+
+### checkout can pull
+
+We already have a checkout command, this RFC proposes that checkout will do the default of `qri pull` if a user tries to checkout a dataset they don't have.
+
+```
+$ qri checkout --help
+Create a linked directory and write dataset files to that directory. 
+
+If the dataset you are checking out isn't in the qri database, the latest 
+version of the dataset will be pulled from the network first. To prevent network
```

Comment:
assuming this also works for a specified hash, if one is given, not just the most recent.

Reply:
_kiiiiinda_. We'll need to land and rely on dscache before we can get there. Currently you can only checkout HEAD